browser/ui/views/page_action/brave_page_action_icon_container_view.cc

```diff
@@ -7,34 +7,47 @@
 
 #include "base/check_is_test.h"
 #include "brave/browser/ui/page_action/brave_page_action_icon_type.h"
+#include "brave/components/brave_player/buildflags/buildflags.h"
 #include "brave/components/playlist/common/features.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/ui/browser.h"
 #include "chrome/browser/ui/views/page_action/page_action_icon_params.h"
 #include "ui/base/metadata/metadata_impl_macros.h"
 
+#if BUILDFLAG(ENABLE_BRAVE_PLAYER)
+#include "brave/components/brave_player/features.h"
+#endif
+
 namespace {
 
 PageActionIconParams& ModifyIconParamsForBrave(PageActionIconParams& params) {
-  if (!base::FeatureList::IsEnabled(playlist::features::kPlaylist)) {
-    return params;
-  }
-
-  if (!params.browser) {
+  // Add actions for Brave
+  if (base::FeatureList::IsEnabled(playlist::features::kPlaylist)) {
     // Browser could be null if the location bar was created for
     // PresentationReceiverWindowView.
-    return params;
+    if (params.browser && params.browser->is_type_normal() &&
+        !params.browser->profile()->IsOffTheRecord()) {
+      // Insert Playlist action before sharing hub or at the end of the vector.
+      params.types_enabled.insert(
+          base::ranges::find(params.types_enabled,
+                             PageActionIconType::kSharingHub),
+          brave::kPlaylistPageActionIconType);
+    }
   }
 
-  // Add actions for Brave
-  if (params.browser->is_type_normal() &&
-      !params.browser->profile()->IsOffTheRecord()) {
-    // Insert Playlist action before sharing hub or at the end of the vector.
-    params.types_enabled.insert(
-        base::ranges::find(params.types_enabled,
-                           PageActionIconType::kSharingHub),
-        brave::kPlaylistPageActionIconType);
+#if BUILDFLAG(ENABLE_BRAVE_PLAYER)
+  if (base::FeatureList::IsEnabled(brave_player::features::kBravePlayer)) {
+    if (params.browser && params.browser->is_type_normal()) {
```

Comment:
should this also check OTR like in https://github.com/brave/brave-core/pull/20730/files#diff-76bbcd0d61753b544d7d923e7d4d487abefe1adc10af4f9ab7073200417b4273R29 ?

Reply:
I'm not sure. Are we going to disable the feature for OTR?

Reply:
do we want it for OTR @pes10k ? 

if so do you want the chrome://player to open in the OTR window, or do you want it to open in a normal window like what happens with settings?

Reply:
If i understand the question, theres shouldn't be anyway for OTR to interact with this feature.  The Brave-player feature can only be accessed from youtube.com, and unless youtube.com starts sending the `Request-OTR: 1` header (not impossible but doesn't seem likely anytime soon), there wont be any way to get into this feature from an OTR session

Reply:
Sorry, I think I'm not following the conversation well. I believe we were talking about secrete/tor window. `Request-OTR: 1` header is used for `Requset-OTR` tab right? Then, it's a little bit different from the OTR window we were talking about. If we want to disable this feature on `Request-OTR tab`, that should be handled else where. So, the question left is if we should disable this entirely from secrete/tor window. To me, opening a new web ui tab from the same window doesn't seem to harm the privacy.

Reply:
Gotcha, yes, my misunderstanding then.  I thought folks were asking about a Request-OTR tab, but I understand the question correctly now.  Sorry, my error.

I dont think we should disable this for private/tor windows, since the threat model here is unrelated to local storage, and the feature (mostly) is unrelated to how local storage is handled.

The feature should would work exactly the same in a private/tor window as it would in a standard window, but with no permanent record of the "brave-viewer" use persisted to disk / history / cache once the window was closed.

If this is difficult to implement, to push back the MVP / v1 signifigantly, then lets discuss more.  But, if the above description sounds implementable w/o much difficulity, then thats the ideal

Reply:
> The feature should would work exactly the same in a private/tor window as it would in a standard window, but with no permanent record of the "brave-viewer" use persisted to disk / history / cache once the window was closed.

I believe this is guaranteed by the current architecture.